pom.xml

```diff
@@ -53,6 +54,28 @@
             <artifactId>mailer</artifactId>
             <!-- TODO: should be transformed into a optional dependency -->
         </dependency>
+        <dependency>
+            <groupId>javax.mail</groupId>
+            <artifactId>javax.mail-api</artifactId>
+            <version>1.5.6</version>
+        </dependency>
+        <dependency>
+            <groupId>org.kohsuke</groupId>
+            <artifactId>access-modifier-annotation</artifactId>
+            <scope>provided</scope>
+            <!-- Use version preferred by jenkins-core -->
+        </dependency>
+        <dependency>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-annotations</artifactId>
+            <scope>provided</scope>
+            <!-- Use version preferred by jenkins-core -->
+        </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>1.3.2</version>
+        </dependency>
```

Comment:
These shouldn't be needed with recent versions of the plugin parent POM (which you are adopting above).

Reply:
It caused this addition :) maven claimed that this plugin "directly" required some versions of the stuff, different than what jenkins-core wanted.

Reply:
By now, removing two of these went without complaints. But it still wanted some specific (newest of those found) annotation-api : 
````
[INFO] Ignoring requireUpperBoundDeps in com.google.code.findbugs:jsr305
[ERROR] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for javax.annotation:javax.annotation-api:1.2 paths to dependency are:
+-org.jvnet.hudson.plugins:instant-messaging:1.50-SNAPSHOT
  +-org.jenkins-ci.main:maven-plugin:3.19
    +-org.apache.maven:maven-embedder:3.8.3
      +-javax.annotation:javax.annotation-api:1.2
and
+-org.jvnet.hudson.plugins:instant-messaging:1.50-SNAPSHOT
  +-org.jenkins-ci.main:maven-plugin:3.19
    +-org.apache.maven.resolver:maven-resolver-impl:1.7.3
      +-javax.annotation:javax.annotation-api:1.3.2
and
+-org.jvnet.hudson.plugins:instant-messaging:1.50-SNAPSHOT
  +-org.jenkins-ci.main:maven-plugin:3.19
    +-org.eclipse.sisu:org.eclipse.sisu.plexus:0.3.5
      +-javax.annotation:javax.annotation-api:1.2
]
````

Reply:
This is really a bug in `maven-plugin`, but to work around it here I think the best strategy is to add an entry to the `dependencyManagement` section for `javax.annotation:javax.annotation-api:1.3.2`. That essentially instructs Maven to choose the latest version (1.3.2) rather than the "nearest" version (1.2) when resolving this dependency. Essentially Enforcer only exists to force you to write this manual workaround for Maven's lack of a "latest" resolution strategy. The key advantage to doing this in a `dependencyManagement` section is that this doesn't force the dependency to be pulled in if it isn't otherwise needed; rather, it only provides "advice" to Maven about how to resolve the dependency.

Reply:
Hm, thanks - I did not recognize that, although some examples suggested adding essentially same tags into two locations (without a version in the latter).